File: 950.py
class Solution:
    def deckRevealedIncreasing(self, deck: List[int]) -> List[int]:
        deck.sort()
        result = [0] * len(deck)
        idx = list(range(len(deck)))
        
        for card in deck:
            poppedIdx = idx.pop(0)  
            result[poppedIdx] = card   
            if idx:               
                idx.append(idx.pop(0))  
        
        return result

File: 950.cpp
class Solution {
public:
    vector<int> deckRevealedIncreasing(vector<int>& deck) {
        sort(deck.begin(), deck.end());
        vector<int> ans(deck.size());
        queue<int> q;
        for (int i=0; i<deck.size(); i++) q.push(i);
        for (int card : deck){
            ans[q.front()] = card;
            q.pop();
            if (!q.empty()){
                q.push(q.front());
                q.pop();
            }
        }
        return ans;
    }
};